eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/LegacyStorage.php

```diff
@@ -212,23 +212,47 @@ public function countImageReferences($uri)
         $connection = $this->getConnection();
 
         $selectQuery = $connection->createSelectQuery();
-        $selectQuery->select(
-            $selectQuery->expr->count(
-                $connection->quoteColumn('id')
-            )
-        )->from(
+        $selectQuery
+            ->select(
+                $selectQuery->expr->count(
+                    $connection->quoteColumn('id')
+                )
+            )->from(
+                $connection->quoteTable('ezimagefile')
+            )->where(
+                $selectQuery->expr->like(
+                    $connection->quoteColumn('filepath'),
+                    $selectQuery->bindValue($path . '%')
+                )
+            );
+
+        $statement = $selectQuery->prepare();
+        $statement->execute();
+
+        return (int)$statement->fetchColumn();
+    }
+
+    public function isImageReferenced(string $uri): bool
+    {
+        $path = $this->redecorator->redecorateFromSource($uri);
+
+        $connection = $this->getConnection();
+
+        $selectQuery = $connection->createSelectQuery();
+        $selectQuery->select(1)
+        ->from(
             $connection->quoteTable('ezimagefile')
         )->where(
-            $selectQuery->expr->like(
+            $selectQuery->expr->eq(
```

Comment:
It results in a different query than the one used for counting (`like` vs `equality`). Is this intentional?

Reply:
Yes. Thats what make the biggest difference and have whole second point in PR description ;)

Reply:
To be precise, in this case always whole path to image is provided, so there is no need to use `LIKE` and expect something else.